generators/generator-base-private.js

```diff
@@ -1353,44 +1353,69 @@ module.exports = class extends Generator {
      * @param {*} options
      */
     getJDBCUrl(databaseType, options = {}) {
+        if (options.protocol) {
+            throw new Error("option 'protocol' is not accepted when asking for a jdbc url, use getDBCUrl() instead");
+        }
+        options.protocol = 'jdbc';
+        return this.getDBCUrl(databaseType, options);
+    }
+
+    /**
+     * Returns the R2DBC URL for a databaseType
+     *
+     * @param {string} databaseType
+     * @param {*} options
+     */
+    getR2DBCUrl(databaseType, options = {}) {
+        if (options.protocol) {
+            throw new Error("option 'protocol' is not accepted when asking for an r2dbc url, use getDBCUrl() instead");
```

Comment:
I don't think `getDBCUrl()` function should be advertised and be public: templates and subgenerators should only use `getR2DBCUrl()` and `getJDBCUrl()`.

So I'd rather change the signature of the `getDBCUrl()` function that way:
`getDBCUrl(databaseType, protocol, options = {})`. 
Then you don't need to check `options`.

Reply:
Thanks! Refactored and cleaned up method signature, not sure about the "advertised and public" part, please advise?

I also cleaned up other usages that you pointed out.